File: chap12/fig12-16.R
library(dplyr)
library(ggplot2)

d <- read.csv('input/data-2Dmesh.csv', header=FALSE)
I <- nrow(d)
J <- ncol(d)

fit_MAP <- readRDS('output/result-model12-9.RDS')
f_est <- fit_MAP$draws('fp', format='matrix') %>% 
  matrix(., I, J)

png('output/fig12-16.png', pointsize=72, width=2400, height=2100)
persp(1:I, 1:J, f_est, theta=55, phi=40, expand=0.5, border='black', col='grey95',
  xlab='Plate Row', ylab='Plate Column', zlab='f', ticktype='detailed', lwd=2)
dev.off()

File: chap12/fig12-16.py
import pandas
import cmdstanpy
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import cm

d = pandas.read_csv('input/data-2Dmesh.csv', header=None).to_numpy()
I, J = d.shape

fit_MAP = cmdstanpy.from_csv('output/result-model12-9')
f_est = fit_MAP.stan_variable(var='fp').reshape((I, J), order='F')

fig, ax = plt.subplots(subplot_kw={"projection": "3d"})
X, Y = np.meshgrid(np.arange(1,I+1), np.arange(1,J+1))
surf = ax.plot_surface(X, Y, f_est.T, cmap=cm.coolwarm, linewidth=0, antialiased=False)                       
ax.set_xlabel('Plate Row')
ax.set_ylabel('Plate Column')
ax.set_zlabel('f')
ax.view_init(40, -30)
fig.savefig('output/fig12-16.py.png', dpi=200)
